### ISSUE TITLE
[sqs] SqsQueueMetricFactory using incorrect metric

### ISSUE THREAD

**Issue by bdgerszewski** [cdklabs/cdk-monitoring-constructs#109]
### Version

v1.4.0

### Steps and/or minimal code example to reproduce

https://github.com/cdklabs/cdk-monitoring-constructs/blame/be6c84982745015ba130b34207197a9531d1d6b5/lib/monitoring/aws-sqs/SqsQueueMetricFactory.ts#L28

`this.queue.metricNumberOfMessagesSent` should not be used for `metricIncomingMessageCount`. 
`this.queue.metricNumberOfMessagesReceived` should be used instead.

### Expected behavior

An alarm on "incoming" messages that depends on SqsQueueMetricFactory::metricIncomingMessageCount should yield an alarm on incoming message count.

### Actual behavior

The resulting alarm is on sent message count

**Comment by echeung-amzn** [comment]
I see a few issues here:

- `NumberOfMessagesSent` captures the number of messages sent to the queue, but does _not_ capture those automatically sent to a DLQ ([ref](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-dead-letter-queues.html#sqs-dlq-number-of-messages))
- `NumberOfMessagesReceived` captures the number of messages processed by `ReceiveMessage` action ([ref](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-available-cloudwatch-metrics.html)), so it doesn't necessarily reflect messages that are just sitting in the queue that have yet to be processed

Realistically, using `ApproximateNumberOfMessagesVisible` is probably the ideal way of monitoring a queue's size.

Can you expand a bit more on your use case?

**Comment by bdgerszewski** [comment]
Ahhh, you've figured it out for me. My use case is monitoring the size of a DLQ. I've already changed to using `ApproximateNumberOfMessagesVisible`, but thought I'd raise an issue for this as it seemed wrong. I suppose the code I linked isn't inherently wrong, but seems off in its use here: https://github.com/cdklabs/cdk-monitoring-constructs/blob/be6c84982745015ba130b34207197a9531d1d6b5/lib/monitoring/aws-sqs/SqsQueueMonitoringWithDlq.ts#L106-L107. Not sure what the resolution ought to be here, perhaps the most appropriate change would be to change a property name or add some documentation.
